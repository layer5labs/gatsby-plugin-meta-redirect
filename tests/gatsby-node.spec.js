const { exists, remove, readFile } = require('fs-extra');
const { onPostBuild } = require('../gatsby-node');

describe('onPostBuild', () => {
  const tempFolderPath = './public';

  const assertRedirectFile = async (redirects, expectedPath) => {
    await onPostBuild({
      store: {
        getState: () => ({
          redirects,
          program: {
            directory: './'
          }
        })
      }
    });

    expect(await exists(expectedPath)).toBe(true);
    expect(await readFile(expectedPath, 'utf-8')).toMatchSnapshot();
  };

  beforeEach(async () => {
    await remove(tempFolderPath);
  });

  // cleanup
  afterAll(async () => {
    await remove(tempFolderPath);
  });

  it('writes redirects from root', async () => {
    await assertRedirectFile(
      [
        {
          fromPath: '/',
          toPath: '/hello'
        }
      ],
      `${tempFolderPath}/index.html`
    );
  });

  it('writes redirects to root', async () => {
    await assertRedirectFile(
      [
        {
          fromPath: '/hello',
          toPath: '/'
        }
      ],
      `${tempFolderPath}/hello/index.html`
    );
  });

  it('writes deep path redirects', async () => {
    await assertRedirectFile(
      [
        {
          fromPath: '/a/b/c/d',
          toPath: '/x/y/z'
        }
      ],
      `${tempFolderPath}/a/b/c/d/index.html`
    );
  });

  it('handles external redirects', async () => {
    await assertRedirectFile(
      [
        {
          fromPath: '/a/b',
          toPath: 'http://example.com/'
        }
      ],
      `${tempFolderPath}/a/b/index.html`
    );
  });
});
