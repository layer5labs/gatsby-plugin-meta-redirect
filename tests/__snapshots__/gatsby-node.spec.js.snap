// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onPostBuild handles external redirects 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='http://example.com/'\\" />"`;

exports[`onPostBuild writes deep path redirects 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/x/y/z/'\\" />"`;

exports[`onPostBuild writes redirects from root 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/hello/'\\" />"`;

exports[`onPostBuild writes redirects to root 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/'\\" />"`;
